client/components/happychat/autoscroll.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Mixin that will scroll to the bottom of a scrollable container whenever it's rendered.
+ * When the scrollable element is scrolled manually by the user autoscroll is disabled until the
+ * user scrolls back to the bottom of the content.
+ *
+ * To use declare the mixin and then call the `setupAutoscroll( node )` method on your component
+ * where `node` is an html element with scrolling enabled.
+ *
+ * After every update the content will be scrolled to the bottom of the content.
+ */
+export default {
+	componentWillMount() {
+		this._autoscroll_enabled = true;
+		window.addEventListener( 'resize', this.scrollToBottom );
+		this.scrollToBottom();
```

Comment:
We probably don't need to call `scrollToBottom` in `WillMount` (likely won't do anything anyway since the node won't be rendered yet).